spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/ribbon/NacosRule.java

```diff
@@ -33,24 +36,37 @@
 import org.slf4j.LoggerFactory;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.commons.util.InetUtils;
 import org.springframework.util.CollectionUtils;
 
 /**
  * Supports preferentially calling the ribbon load balancing rules of the same cluster
  * instance.
  *
  * @author itmuch.com
+ * @author HH
  */
 public class NacosRule extends AbstractLoadBalancerRule {
 
 	private static final Logger LOGGER = LoggerFactory.getLogger(NacosRule.class);
 
+	private static final String IPV4_REGEX = "^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\."
```

Comment:
use ((2(5[0-5]|[0-4]\d))|[0-1]?\d{1,2})(\.((2(5[0-5]|[0-4]\d))|[0-1]?\d{1,2})){3} instead of it, you can refer to https://c.runoob.com/front-end/854/

Reply:
done